### ISSUE TITLE
Compile error related to igraph 0.7.1

### ISSUE THREAD

**Issue by ndanger000** [shadow/shadow#263]
When I compile shadow against igraph 0.7.1, I get the following (lots of custom install locations for various tools):

```
[ndanner@petaltail shadow]$ ./setup build --include /home/ndanner/opt/include --include /home/ndanner/opt/share/llvm/cmake --include $HOME/opt/include/glib-2.0 --include /share/apps/CENTOS6/gcc/4.8.4/include --include /home/ndanner/opt/igraph/include --library /home/ndanner/opt/igraph/lib --library $HOME/opt/lib  --clean -j 1
...
-- Found components for IGRAPH
-- IGRAPH_INCLUDES = /home/ndanner/opt/igraph/include/igraph
-- IGRAPH_LIBRARIES = /home/ndanner/opt/igraph/lib/libigraph.so
...
...
[ 53%] Building C object src/CMakeFiles/shadow.dir/topology/shd-dns.c.o
[ 54%] Building C object src/CMakeFiles/shadow.dir/topology/shd-path.c.o
[ 55%] Building C object src/CMakeFiles/shadow.dir/topology/shd-topology.c.o
/home/ndanner/src/shadow/src/topology/shd-topology.c:731:100: error: too few
      arguments to function call, expected 9, have 7
  ...igraph_vss_vector(&dstVertexIndexSet), top->edgeWeights, IGRAPH_OUT);
                                                                        ^
/home/ndanner/opt/igraph/include/igraph/igraph_paths.h:92:1: note: 
      'igraph_get_shortest_paths_dijkstra' declared here
int igraph_get_shortest_paths_dijkstra(const igraph_t *graph,
^
1 error generated.
make[2]: *** [src/CMakeFiles/shadow.dir/topology/shd-topology.c.o] Error 1
make[1]: *** [src/CMakeFiles/shadow.dir/all] Error 2
make: *** [all] Error 2
2015-02-19 14:26:42,069 INFO make returned 2
2015-02-19 14:26:42,070 ERROR Non-zero return code from make.
2015-02-19 14:26:42,070 INFO returning code '2'
```

**Comment by robgjansen** [comment]
Thanks for reporting. The igraph team loves changing their function signatures, but its a good thing [they were nice enough to define version flags](https://github.com/igraph/igraph/issues/728) so I can deal with multiple version properly.

**Comment by ndanger000** [comment]
I was just looking at this.

The "stock download" for 0.7.1 (i.e., 
http://igraph.org/nightly/get/c/igraph-0.7.1.tar.gz) doesn't have the 
IGRAPH_VERSION_{MAJOR,MINOR,PATCH} flags, just IGRAPH_VERSION.  And 
that's where things are breaking down...

	- Norman

On 2/19/15 4:40 PM, Rob Jansen wrote:
> Thanks for reporting. The igraph team loves changing their function signatures, but its a good thing [they were nice enough to define version flags](https://github.com/igraph/igraph/issues/728) so I can deal with multiple version properly.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/shadow/shadow/issues/263#issuecomment-75142054
>

-- 
Norman Danner - ndanner@wesleyan.edu - http://ndanner.web.wesleyan.edu
Department of Mathematics and Computer Science - Wesleyan University

**Comment by robgjansen** [comment]
It apparently is merged into the igraph `release-0.7` branch. Are you able to try that instead of the tarball?

**Comment by ndanger000** [comment]
Alas, not easily.

I already tried that, discovered that it needs yacc to be built this 
way, and the cluster I'm on doesn't have yacc.  And I'm getting sort of 
sick of building an entire operating system from source in order to run 
shadow...

I'll see if I can get yacc installed, and then see what doesn't work.

	- Norman

On 2/19/15 5:30 PM, Rob Jansen wrote:
> It apparently is merged into the igraph `release-0.7` branch. Are you able to try that instead of the tarball?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/shadow/shadow/issues/263#issuecomment-75152674
>

-- 
Norman Danner - ndanner@wesleyan.edu - http://ndanner.web.wesleyan.edu
Department of Mathematics and Computer Science - Wesleyan University

**Comment by robgjansen** [comment]
Another easier option here is to just edit `shadow/src/topology/shd-topology.c` and replace lines 720-736 with the one line therein that should be called for igraph version >0.7.0, i.e.:

```
result = igraph_get_shortest_paths_dijkstra(&top->graph, &resultPaths, NULL,
            srcVertexIndex, igraph_vss_vector(&dstVertexIndexSet), top->edgeWeights, IGRAPH_OUT, NULL, NULL);
```

I'm trying to hack together a general fix, but it will be messy...
